### ISSUE TITLE
Using a Haiku network with state for the provided (JAX-) DQN agent

### ISSUE THREAD

**Issue by magnusdk** [deepmind/acme#160]
Hi :),

I wanted to experiment with using a Haiku MobileNetV1 network for the `acme.agents.jax.dqn.agent.DQN` agent. MobileNet uses batch normalization, and therefore has internal state (e.g., uses `haiku.transform_with_state`). I can’t find a way to make this work out of the box with the provided DQN agent, since the implementation expects the network to not use state. Am I missing something?

If not, then I will attempt to create my own version of the agent. My thought is to always assume that the network uses state, since this is most general. This might lock the API in on using Haiku for networks though, which is fine for me, but you guys might have a different philosophy.

Thanks!

**Comment by nikolamomchev** [comment]
Hi @magnusdk,

Thanks for reporting this. Indeed DQN currently only supports feed-forward networks with no state. We have some changes in the pipeline which should make this possible but for the time being your best option is to fork DQN and modify it to support the network type you care about.
